flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/TieredStorageProducerClient.java

```diff
@@ -119,7 +149,60 @@ private void writeAccumulatedBuffers(
     private void writeAccumulatedBuffer(
             TieredStorageSubpartitionId subpartitionId, Buffer accumulatedBuffer)
             throws IOException {
-        // TODO, Try to write the accumulated buffer to the appropriate tier. After the tier is
-        // decided, then write the accumulated buffer to the tier.
+        updateStatistics(accumulatedBuffer);
+        Buffer compressedBuffer = compressBufferIfPossible(accumulatedBuffer);
+
+        if (currentSubpartitionTierAgent[subpartitionId.getSubpartitionId()] == null) {
+            chooseStorageTierToStartSegment(subpartitionId);
+        }
+
+        boolean isSuccess =
+                currentSubpartitionTierAgent[subpartitionId.getSubpartitionId()].write(
```

Comment:
The semantic of `TierProducerAgent#write` return value needs to be documented. E.g., after a write failure, do we allow writing to the segment again?

Reply:
Added docs for the `write` and added docs for `TierProducerAgent` API.